Sources/Datadog/DatadogInternal/DatadogCoreProtocol.swift

```diff
@@ -11,21 +11,7 @@ public internal(set) var defaultDatadogCore: DatadogCoreProtocol = NOOPDatadogCo
 /// A Datadog Core holds a set of features and is responsible of managing their storage
 /// and upload mechanism. It also provides a thread-safe scope for writing events.
 public protocol DatadogCoreProtocol {
-    /// Registers a feature by its name and configuration.
-    ///
-    /// - Parameters:
-    ///   - featureName: The feature name.
-    ///   - storage: The feature's storage configuration.
-    ///   - upload: The feature's upload configuration.
-    func registerFeature(named featureName: String, storage: FeatureStorageConfiguration, upload: FeatureUploadConfiguration)
-
-    /// Returns a Feature scope by its name.
-    ///
-    /// - Parameter featureName: The feature's name.
-    /// - Returns: The scope for feature that previously registered, `nil` otherwise.
-    func scope(forFeature featureName: String) -> FeatureScope?
-
-    // MARK: V1 interface
```

Comment:
As discussed, we will bring it back when `FeatureStorageConfiguration` and `FeatureUploadConfiguration` can become `public` APIs. It's not possible now, as they depend on things that will drag a lot of other types if made public. I should be able to do something with this when both RUM and Tracing are also migrated to V2 config.